### ISSUE TITLE
Performance issue for extension modules

### ISSUE THREAD

**Issue by anakin1028** [wifiphisher/wifiphisher#841]
@sophron we have talked about this a little in #742 and I feel it worth to open an issue for tracking this problem. I have done one experiment today: Compare the sniffing frames from our tool, say `wlan0` and the other one is from another monitor interface, say `wlan2` by `wireshark`. The following is the result:

`Our tool`: you can see the following picture, `Wifiphisher` only handles 30 packets in one second:
![wifiphisher_sniffer](https://user-images.githubusercontent.com/6182530/34907234-5ba91598-f8b6-11e7-8180-35f9eb689947.png)

`For dedicate WLAN interface sniffing in wireshark`
* starting packet number: 2236
![dedicate_sniffer_2236](https://user-images.githubusercontent.com/6182530/34907259-bedc661a-f8b6-11e7-88ae-83252c340568.png)
* end packet number: 2655
![dedicate_sniffer_2655](https://user-images.githubusercontent.com/6182530/34907269-f02a6442-f8b6-11e7-8179-779f7b3f0900.png)

This means in 1 second there are about 400 frames on the air; however our tool only catches about 30 frames. This is because our extension frame work sniff a frame at one time and after process all the works by each extension modules, then we sniff the next one:

```python
while self._should_continue:
            dot11.sniff(
                iface=self._interface,
                prn=self._process_packet,
                count=1,
                store=0,
                stop_filter=self._stopfilter)
```
@sophron @blackHatMonkey I feel maybe we should dedicate one process to sniff the frames on the air and store them in the queues (we can define high priority queue (special frames: like EAPOL, BEACONs for target AP) and normal priority queue (all others)) and then extension modules get the frames from those queues. What do you think?

**Comment by blackHatMonkey** [comment]
@anakin1028 I think it is a very good idea to use priority queues for this. I can defiantly help implement or review this 👍 .

**Comment by sophron** [comment]
Thank you @anakin1028 for researching this! 

If we instruct Scapy to capture all the frames without any processing, do we capture all the frames then? If yes, then indeed the problem is what you described.

**Comment by blackHatMonkey** [comment]
This is somewhat implemented in #842.
